bijection-util/src/main/scala/com/twitter/bijection/twitter_util/JavaFutureConverter.scala

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2012 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.bijection.twitter_util
+
+import java.util.concurrent.{ Future => JFuture }
+
+import com.twitter.util._
+
+/**
+ * Base class for converting java futures to twitter futures
+ */
+abstract class JavaFutureConverter {
+  def apply[T](javaFuture: JFuture[T]): Future[T]
+}
+
+/**
+ * Converter based on a [[FuturePool]] which will create one thread per future.
+ * To favor if there aren't too many futures to convert and one cares about latency.
```

Comment:
maybe worth adding a note that we use an unbounded threadpool to avoid potential deadlocks.